query/language/negation.feature

```diff
@@ -605,5 +605,31 @@ Feature: Negation Resolution
     Then answer size is: 3
 
 
+  Scenario: Negation inputs are handled correctly
+    Given connection open schema transaction for database: typedb
+    Given typeql write query
+      """
+      insert
+      $x isa person;
+      $c isa company;
+      $e isa employment, links (employee: $x, employer: $c);
+      """
+    Given transaction commits
+    Given connection open read transaction for database: typedb
+    When get answers of typeql read query
+      """
+      match
+        not { # Must be in a negation so that `$x` is not selected for the entire match stage.
+          $x isa person;
+          not {
+            (employee: $x);
+            # The planner prefers ordering `$_ role-name employee` (which is free) before `$_ links $_:$x` (which has to hit disk).
+            # If `$x` is not marked as input to the negation during lowering, it is not selected by the role name step, and therefore removed from the row.
+            # The links executor then crashes as it expects `$x` to be bound.
+          };
+        };
+      """
```

Comment:
Do you want to verify the answer size in a `then` just to make sure the executor actually does something?

Reply:
It's a negation, it can only ever output one zero-width row.

Reply:
Makes sense